octoprint_mrbeam/static/js/convert.js

```diff
@@ -616,25 +616,56 @@ $(function () {
             self._update_job_summary();
         });
 
+        self.showMaterialCompatibility = function () {
+            let customMaterialsLaserModels = [];
+            let customMaterials = self.custom_materials();
+            // Get all laser models used while saving custom materials
+            for (let materialKey in customMaterials) {
+                if (customMaterials.hasOwnProperty(materialKey)) {
+                    let m = customMaterials[materialKey];
+                    if (('laser_model' in m) && !customMaterialsLaserModels.includes(m.laser_model)) {
+                        customMaterialsLaserModels.push(m.laser_model);
+                    } else if (!('laser_model' in m) &&
+                        ('laser_type' in m) &&
+                        m.laser_type === 'MrBeamII-1.0' &&
+                        !customMaterialsLaserModels.includes('0')) {
+                        customMaterialsLaserModels.push('0');
+                    }
```

Comment:
I believe that - instead of supporting the legacy materials which don't have the `laser_type` key, we should be doing a migration of the materials in the backend + (especially since it would be backwards compatible) - and remove this extra `else` if the key is not present in the settings.

If the key is not found, I would prefer to show a warning that it is from an unknown laserhead.

N.B. Keep in mind that the import function for the custom materials should also do this migration - or not as we are not sure if it was from a MrB II or a DC device.